pkg/vmprovider/providers/vsphere2/vmlifecycle/bootstrap.go

```diff
@@ -57,7 +57,8 @@ func DoBootstrap(
 
 	bootstrap := vmCtx.VM.Spec.Bootstrap
 	if bootstrap == nil {
-		return nil
+		// For backwards compat we have to keep going and default to LinuxPrep.
+		bootstrap = &vmopv1.VirtualMachineBootstrapSpec{}
```

Comment:
I assume it will be safe to augment this once we introduce knowledge of the image's reported OS and default to Sysprep if Windows is detected?

Reply:
Yes, as long as we want to continue with the implied GOSC (IDK if the v1a1 behavior was intentional but it was surprising to me when tracking down e2e failures some months back)

Reply:
It was not intentional so much as a byproduct of being first and once other things were added, it just...defaulted.